### PR TITLE
contrib/go-redis/redis: Enable tracing of Pipelined and TxPipelined

### LINKED ISSUE #720: contrib/go-redis/redis: Support tracing of client.Pipelined and client.TxPipelined
I'm using the `TxPipelined(fn func(Pipeliner) error) ([]Cmder, error) ` function of `go-redis` but not receiving traces for any of the calls made through the pipeline. 

I was able to get these calls traced by adding the following code to the `contrib/go-redis/redis/redis.go` file:

```
func (c *Client) Pipelined(fn func(redis.Pipeliner) error) ([]redis.Cmder, error) {
	return c.Pipeline().Pipelined(fn)
}

func (c *Client) TxPipelined(fn func(redis.Pipeliner) error) ([]redis.Cmder, error) {
	return c.TxPipeline().Pipelined(fn)
}

func (c *Client) TxPipeline() redis.Pipeliner {
	return &Pipeliner{c.Client.TxPipeline(), c.params, c.Client.Context()}
}

func (c *Pipeliner) Pipelined(fn func(redis.Pipeliner) error) ([]redis.Cmder, error) {
	if err := fn(c); err != nil {
		return nil, err
	}
	cmds, err := c.Exec()
	_ = c.Close()
	return cmds, err
}
```

If that looks good I'd be happy to write a test and open a PR for it :)

### PR DESCRIPTION
**Pull request by JDiPierro**
Enable tracing of Pipelined and TxPipelined calls.

Fixes #720

### CODE REVIEW

File: contrib/go-redis/redis/redis.go [hunk 1]
```diff
@@ -133,6 +148,16 @@ func commandsToString(cmds []redis.Cmder) string {
 	return b.String()
 }
 
+// Pipelined executes a function parameter to build a Pipeline and then immediately executes the built pipeline.
+func (c *Pipeliner) Pipelined(fn func(redis.Pipeliner) error) ([]redis.Cmder, error) {
+	if err := fn(c); err != nil {
+		return nil, err
+	}
+	cmds, err := c.Exec()
+	_ = c.Close()
```
Comment (gbbr) [inline contrib/go-redis/redis/redis.go:157]: Why are we ignoring this error?

**Review by gbbr — COMMENTED** [review]
Thanks! One question

File: contrib/go-redis/redis/redis.go (hunk 1 above)
Comment (JDiPierro) [reply contrib/go-redis/redis/redis.go:157]: `Close` will actually never return an error: https://github.com/go-redis/redis/blob/master/pipeline.go#L71

I would guess they wanted it to conform to the `Closer` interface in the `io` standard library.

File: contrib/go-redis/redis/redis.go [hunk 2]
```diff
@@ -133,6 +148,16 @@ func commandsToString(cmds []redis.Cmder) string {
...
+	return cmds, err
```
Comment (knusbaum) [inline contrib/go-redis/redis/redis.go:158]: I know this was copied directly from `go-redis`, but I think we can clean it up a little. 
```suggestion
	defer c.Close()
	return c.Exec()
```

**Review by knusbaum — CHANGES_REQUESTED** [review]
One minor comment. Otherwise, this looks great.

File: contrib/go-redis/redis/redis.go (hunk 2 above)
Comment (JDiPierro) [reply contrib/go-redis/redis/redis.go]: Sounds great, thanks for the suggestion :)

**Review by knusbaum — APPROVED** [review]
LGTM. Thanks, @JDiPierro

**Pull request merged**